ex_schemas/schema_v3.json

```diff
@@ -0,0 +1,496 @@
+                         "pattern":"duffy_grp"
+                     },
+                     "res_defs":{
+                         "type": "array",
+                         "items" : {
+                             "oneOf" :
+                                 [
+                                   {"$ref": "#/definitions/duffy"}
+                                 ]
+                            }
+                    },
+                      "assoc_creds":{
+                        "description":"contains creds file associated to this resource",
+                        "type":"string"
+                    }
+                 },
+                 "required":["resource_group_name","res_group_type","res_defs","assoc_creds"],
+                 "additionalProperties": true
+         },
+         "libvirt_grp": {
+                 "type" : "object",
+                 "properties" : {
+                     "resource_group_name": {
+                         "type":"string"
+                     },
+                     "res_group_type": {
+                         "type":"string",
+                         "pattern":"libvirt_grp"
+                     },
+                     "res_defs":{
+                         "type": "array",
+                         "items" : {
+                             "oneOf" :
+                                 [
+                                   {"$ref": "#/definitions/libvirt"}
+                                 ]
+                            }
+                    },
+                      "assoc_creds":{
+                        "description":"contains creds file associated to this resource",
+                        "type":"string"
+                    }
+                 },
+                 "required":["resource_group_name","res_group_type","res_defs","assoc_creds"],
+                 "additionalProperties": true
+         },
+         "rackspace": {
+                 "type" : "object",
+                 "properties" : {
+                     "resource_group_name": {
+                         "type":"string"
+                     },
+                     "res_group_type": {
+                         "type":"string",
+                         "pattern":"rackspace"
+                     },
+                     "res_defs":{
+                         "type": "array",
+                         "items" : {
+                             "oneOf" :
+                                 [
+                                   {"$ref": "#/definitions/rax_server"}
+                                 ]
+                            }
+                    },
+                      "assoc_creds":{
+                        "description":"contains creds file associated to this resource",
+                        "type":"string"
+                    }
+                 },
+                 "required":["resource_group_name","res_group_type","res_defs","assoc_creds"],
+                 "additionalProperties": true
+         }
+    }
+}
+
```

Comment:
How does this differ from schema_v2? I'd like to start using v3 for my stuff, is that likely going to work? It's hard to tell without a diff.

Additionally, having a default schema in the config seems like a good idea, unless it's been overridden.